makeStage2.sh

```diff
@@ -39,5 +40,66 @@ cd $usepop/pop/x/src/
 # ways to construct the Poplog system.
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
+# $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p ${BUILD_HOME}/environments
+
+# We need to capture the Poplog environment for each build variant.
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g'
+}
+
+### Xt #########################################################################
+
 $usepop/pop/src/newpop -link -x=-xt -norsv
 
+echo_env "$usepop" > ${BUILD_HOME}/environments/xt
+echo_env "$usepop/pop/.." > ${BUILD_HOME}/environments/xt-cmp
+( cd ${BUILD_HOME}/environments && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xt!g' < xt | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xt/!g' > xt-new )
+
+mkdir -p $usepop/pop/pop-xt
+mkdir -p $usepop/pop/lib/psv-xt
+( cd $usepop/pop/pop; tar cf - . ) | ( cd $usepop/pop/pop-xt; tar xf - )
```

Comment:
As this is such a prevalent idiom, can this be refactored out to a helper function `cpdir` or something  along those lines?

Reply:
I was actually thinking of using `cp -pPr`. I guess it would be easy enough to write a cpFromTo function. Something like:
```
cpFromTo() {
     [ -e $1 ] || exit 1
     [ -e $2 ] || exit 2
     ( cd $1; tar cf - . ) | ( cd $2; tar xf - )
}
```

Reply:
> The only thing to watch out for with cp is that it behaves differently depending on whether or not the target exists (yay!). So it is imperative to know if the target exists (better choice) or does not exist (but check that a previously broken build will not disrupt this assumption).

As in it'll create a new folder if the target doesn't exist, otherwise it'll copy into the existing folder if it does?

Reply:
Yes. By and large we are copying into existing folders, so we should be OK.

Reply:
Drat! In this particular case, it is NOT correct to use `cp -a`. You must precede it with `rm -rf DESTINATION` which creates a danger for scripting boo-boos. So I propose to check that the destination does not exist in the target folder & abort if it does. This will very occasionally require a make clean. I think this is the best answer.

Reply:
After assessing the impact on the scripts I have simply followed your original suggestion and refactored out to a helper function. The ancient idiom remains the least error-prone as far as I can tell.